tests/websocket_new_test.py

```diff
@@ -228,3 +229,245 @@ def error_detector(environ, start_response):
+            ('permessage-deflate; server_max_window_bits; '
+             'client_max_window_bits=9', [15, 9])
+        ]:
+            sock = eventlet.connect(self.server_addr)
+
+            sock.sendall(six.b(self.connect % extension_string))
+            result = sock.recv(1024)
+
+            # The server responds the correct Websocket handshake
+            # print('Extension offer: %r' % extension_string)
+            match = re.match(self.handshake_re, result)
+            self.assertIsNotNone(match)
+            self.assertEqual(1, len(match.groups()))
+
+            offered_parts = [part.strip().lower() for part in extension_string.split(';')]
+            offered_parts_names = [part.split('=')[0].strip() for part in offered_parts]
+            offered_parts_dict = dict(zip(offered_parts_names[1:], vals))
+
+            accepted_ext_parts = match.groups()[0].decode().split('; ')
+            self.assertEqual(accepted_ext_parts[0], 'permessage-deflate')
+            for param, val in (part.split('=') for part in accepted_ext_parts[1:]):
+                self.assertLessThanEqual(int(val), offered_parts_dict[param])
+
+    def test_reject_max_window_bits_out_of_range_13(self):
+        extension_string = ('permessage-deflate; client_max_window_bits=7,'
+                            'permessage-deflate; server_max_window_bits=16, '
+                            'permessage-deflate; client_max_window_bits=16; '
+                            'server_max_window_bits=7, '
+                            'permessage-deflate')
+        sock = eventlet.connect(self.server_addr)
+
+        sock.sendall(six.b(self.connect % extension_string))
+        result = sock.recv(1024)
+
+        # The server responds the correct Websocket handshake
+        # print('Extension offer: %r' % extension_string)
+        match = re.match(self.handshake_re, result)
+        self.assertEqual(match.groups()[0], b'permessage-deflate')
+
+    def test_server_compress_with_context_takeover_13(self):
+        extensions_string = 'permessage-deflate; client_no_context_takeover;'
+        extensions = {'permessage-deflate': {
+            'client_no_context_takeover': True,
+            'server_no_context_takeover': False}}
+
+        sock = eventlet.connect(self.server_addr)
+        sock.sendall(six.b(self.connect % extensions_string))
+        sock.recv(1024)
+        ws = websocket.RFC6455WebSocket(sock, {}, client=True,
+                                        extensions=extensions)
+
+        # Deflated values taken from Section 7.2.3 of RFC 7692
+        # https://tools.ietf.org/html/rfc7692#section-7.2.3
+        ws.send(b'Hello')
+        msg1 = self.get_deflated_reply(ws)
+        self.assertEqual(msg1, b'\xf2\x48\xcd\xc9\xc9\x07\x00')
+
+        ws.send(b'Hello')
+        msg2 = self.get_deflated_reply(ws)
+        self.assertEqual(msg2, b'\xf2\x00\x11\x00\x00')
+
+        ws.close()
+        eventlet.sleep(0.01)
```

Comment:
I fear this small sleep at end of test. What is it for?

Reply:
Honestly, I don't know. When creating these tests, I started from [`test_send_recv_13()`](https://github.com/costasgambit/eventlet/blob/ab2300a5eded021c6b4dd07bcf07f032526d4557/tests/websocket_new_test.py#L113), which I assumed to be correct and uses the same [short sleep](https://github.com/costasgambit/eventlet/blob/ab2300a5eded021c6b4dd07bcf07f032526d4557/tests/websocket_new_test.py#L134). I just copied it, thinking it might be important to allow to finish the cleanup after the test.

Reply:
I see, leave it then. I will rewrite it some boring winter day.